client/src/components/FormBuilderModal/FormBuilderModal.js

```diff
@@ -1,104 +1,71 @@
-      return null;
-    }
-    return (
-      <FormBuilderLoaderComponent
-        fieldHolder={{ className: classnames('modal-body', this.props.bodyClassName) }}
-        actionHolder={{ className: 'modal-footer' }}
-        autoFocus={this.props.autoFocus}
-        schemaUrl={this.props.schemaUrl}
-        onSubmit={this.handleSubmit}
-        onAction={this.props.onAction}
-        onLoadingError={this.handleLoadingError}
-        identifier={this.props.identifier}
-      />
-    );
-  }
-
-  /**
-   * Generates the response part of the Modal
-   *
-   * @returns {Component}
-   */
-  getResponse() {
-    if (!this.state || !this.state.response) {
-      return null;
-    }
-
-    let className = '';
-
-    if (this.state.error) {
-      className = this.props.responseClassBad;
-    } else {
-      className = this.props.responseClassGood;
-    }
-
-    return (
-      <div className={className}>
-        { castStringToElement('span', { html: this.state.response }) }
-      </div>
-    );
-  }
-
-  /**
-   * Removes the response from the state
-   */
-  clearResponse() {
-    this.setState({
-      response: null,
-    });
-  }
-
-  handleLoadingError(schema) {
-    const providesOnLoadingError = this.props.onLoadingError !== noop;
-    if (this.props.showErrorMessage || !providesOnLoadingError) {
-      const error = schema.errors && schema.errors[0];
-      this.setState({
-        response: error.value,
-        error: true,
-      });
+/**
+ * React component for displaying a Form in a Modal using a form schema URL
+ */
+function FormBuilderModal(props) {
+  const {
+    responseClassBad,
+    responseClassGood,
+    onLoadingError,
+    showErrorMessage,
+    onClosed,
+    onSubmit,
+    schemaUrl,
+    bodyClassName,
+    FormBuilderLoaderComponent,
+    children
+  } = props;
+
+  const [response, setResponse] = useState(null);
+  const [error, setError] = useState(null);
+
+  // Register a success message
+  const setSuccess = (message) => {
+    setResponse(message);
+    setError(false);
+  };
+
+  // Register a failure message
+  const setFailure = (message) => {
+    setResponse(message);
+    setError(true);
+  };
+
+  // Clear the response
+  const clearResponse = () => {
+    setResponse(null);
+    setError(false);
+  };
+
+  const handleLoadingError = (schema) => {
+    const providesOnLoadingError = onLoadingError !== noop;
+    if (showErrorMessage || !providesOnLoadingError) {
+      const errorResponse = schema.errors && schema.errors[0];
+      setFailure(errorResponse.value);
     }
     if (providesOnLoadingError) {
-      this.props.onLoadingError(schema);
+      onLoadingError(schema);
     }
-  }
+  };
 
   /**
    * Call the callback for hiding this Modal
    */
-  handleHide() {
-    this.clearResponse();
-    if (typeof this.props.onClosed === 'function') {
-      this.props.onClosed();
+  const handleHide = () => {
+    clearResponse();
+    if (typeof onClosed === 'function') {
```

Comment:
```suggestion
    if (onClosed) {
```
Type check via propTypes

Reply:
propTypes doesn't actually affect runtime, I don't think?

Reply:
`proptypes` are a soft check on types. They will throw a console warnings if the prop doesn't match the expected type.

In principal, I think it's fair to tell people "we've defined proptypes to tell you what we expect ... if you ignore them and give us something else, that's on you".

I'm also not opposed to doubling up those checks either.

In this instance, I'm simply replicated the pre-existing behaviour. So the suggestion would somewhat alter the original behaviour of the component.